internal/controllers/analyzer/analyzer.go

```diff
@@ -457,3 +469,60 @@ func (a *Analyzer) isWarning(err string) bool {
 		strings.Contains(err, messages.MsgWarnBrakemanNotRubyOnRailsProject) ||
 		strings.Contains(err, messages.MsgWarnGemfileIsRequiredForBundler)
 }
+
+// getAllVulnerabilitiesWithDetailsJoined will add the default separator of the details and will check if the hash
+// already exists in list of the vulnerabilities, if is duplicated, so, it will only join both details.
+// nolint:funlen,gocyclo // Breaking this function will make it more confusing
+func (a *Analyzer) getAllVulnerabilitiesWithDetailsJoined() (
+	newAnalysisVulnerabilities []analysis.AnalysisVulnerabilities,
+) {
+	for currentIndex := range a.analysis.AnalysisVulnerabilities {
+		currentAV := a.analysis.AnalysisVulnerabilities[currentIndex]
+		exists := false
+		for newIndex := range newAnalysisVulnerabilities {
+			newAV := newAnalysisVulnerabilities[newIndex]
+			if currentAV.Vulnerability.VulnHash == newAV.Vulnerability.VulnHash &&
+				!strings.Contains(newAV.Vulnerability.Details, currentAV.Vulnerability.Details) {
+				exists = true
+				newAnalysisVulnerabilities[newIndex].Vulnerability.Details = fmt.Sprintf("%s\n         %s%s",
+					newAV.Vulnerability.Details, detailsHeaderText, currentAV.Vulnerability.Details)
+				break
+			}
+		}
+		if !exists {
+			currentAV.Vulnerability.Details = fmt.Sprintf("%s%s", detailsHeaderText, currentAV.Vulnerability.Details)
+			newAnalysisVulnerabilities = append(newAnalysisVulnerabilities, currentAV)
+		}
+	}
+	return newAnalysisVulnerabilities
+}
+
+// setCounterOfDetailsDuplicated will check how many details there are by looking
+// for the detailsHeaderText separator constant and will update to add a counter of the details in this vulnerability.
+// nolint:funlen,gocyclo // Breaking this function will make it more confusing
+func (a *Analyzer) setCounterOfDetailsDuplicated(
```

Comment:
We can't join all vulnerabilities details and set the counter in a single step? I think that would be good if we can do both steps in a single loop iteration.